src/hls-adapter.js

```diff
@@ -664,6 +665,7 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
         index: this._playerTracks.filter(track => track instanceof TextTrack).length
       };
       textTrack = new TextTrack(settings);
+      this._nativeTextTracksMap[settings.index] = CEATextTrack;
```

Comment:
_playerTracks and _nativeTextTracksMap are not the same?

Reply:
no. `_playerTracks` it's all tracks (video/audio/text) and not native only

Reply:
Native could only be 608/708?

Reply:
Make sure that https://github.com/kaltura/playkit-js-hls/blob/master/src/hls-adapter.js#L709 is ok as well in case both native and HLS subtitle available it'll not select the native track

Reply:
The captions could be either native or by manifest. Not both of them in the same playback.

Reply:
They actually could be although not the popular use case.
Is that a problem?

Reply:
If couldn't bw both let use _playerTracks with filter with instance of TextTrack

Reply:
@OrenMe we are parsing native tracks only if there are no captions by manifest see https://github.com/kaltura/playkit-js-hls/blob/ad220f8d2af12936618d048a29eb38b10277bc6f/src/hls-adapter.js#L416

Reply:
@Yuvalke `_playerTracks` holds our tracks while `_nativeTextTracksMap` holds the native

Reply:
what is the reason we did this?

Reply:
maybe to prevent duplication as the selection made by lang

Reply:
as we do it now by index we have room for optimization.
@Yuvalke this is what we discussed - pleaselets make sure we have follow up ticket to handle moving to index selection all around.